langchain4j-couchbase/src/main/java/dev/langchain4j/store/embedding/couchbase/CouchbaseEmbeddingStore.java

```diff
@@ -0,0 +1,281 @@
+            String scopeName,
+            String collectionName,
+            String searchIndex,
+            Integer dimensions
+    ) {
+        this.cluster = Cluster.connect(clusterUrl, username, password);
+        Bucket bucket = cluster.bucket(bucketName);
+        bucket.waitUntilReady(Duration.ofSeconds(10));
+
+        this.collection = bucket.scope(scopeName).collection(collectionName);
+        this.searchIndex = searchIndex;
+
+        if (cluster.searchIndexes().getAllIndexes().stream().noneMatch(i -> i.name().equals(searchIndex))) {
+            HashMap<String, Object> sourceParams = new HashMap<>();
+            HashMap<String, Object> params = new HashMap<>();
+            params.put("doc_config", docConfig());
+            params.put("mapping", mapping(dimensions));
+            HashMap<String, Object> planParams = new HashMap<>();
+            SearchIndex index = new SearchIndex(
+                    null,
+                    searchIndex,
+                    "fulltext-index",
+                    params,
+                    null,
+                    bucketName,
+                    sourceParams,
+                    "gocbcore",
+                    planParams
+            );
+            cluster.searchIndexes().upsertIndex(index);
+        }
+    }
+
+    private Map<String, Object> mapping(Integer dimensions) {
+        Map<String, Object> mapping = new HashMap<>();
+        mapping.put("default_analyzer", "standard");
+        mapping.put("default_datetime_parser", "dateTimeOptional");
+        mapping.put("default_field", "_all");
+        Map<String, Object> defaultMapping = new HashMap<>();
+        defaultMapping.put("dynamic", true);
+        defaultMapping.put("enabled", false);
+        mapping.put("default_mapping", defaultMapping);
+        mapping.put("default_type", "_default");
+        mapping.put("docvalues_dynamic", false);
+        mapping.put("index_dynamic", true);
+        mapping.put("store_dynamic", true);
+        mapping.put("type_field", "_type");
+        mapping.put("types", types(dimensions));
+        return mapping;
+    }
+
+    private Map<String, Object> types(Integer dimensions) {
+        Map<String, Object> types = new HashMap<>();
+        Map<String, Object> type = new HashMap<>();
+
+        type.put("dynamic", false);
+        type.put("enabled", true);
+        type.put("properties", properties(dimensions));
+
+        types.put(String.format("%s.%s", collection.scopeName(), collection.name()), type);
+        return types;
+    }
+
+    private Map<String, Object> properties(int dimensions) {
+        Map<String, Object> props = new HashMap<>();
+        props.put("vector", embedding(dimensions));
+        props.put("metadata", metadata());
+        props.put("text", text());
+        return props;
+    }
+
+    private Map<String, Object> text() {
+        Map<String, Object> text = new HashMap<>();
+        text.put("enabled", true);
+        List fields = new ArrayList();
```

Comment:
```suggestion
        List<Map<String, Object>> fields = new ArrayList<>();
```
same for all similar cases below